lib/diyepw/diyepw/convert_noaa_isd_lite_to_amy_epw_files.py

```diff
@@ -0,0 +1,172 @@
+        of the pandas.DataFrame.interpolate() method's "limit" argument for more details. Basically, if a sequence of
+        fields up to the length defined by this argument are missing, those missing values will be interpolated linearly
+        using the values of the fields immediately preceding and following the missing field(s). If a sequence of fields
+        is longer than this limit, then those fields' values will be imputed instead (see max_records_to_impute)
+    :param max_records_to_impute: The maximum number of records to impute. For groups of missing records larger than the
+                            limit set by --max-records-to-interpolate, we replace the missing values using the
+                            average of the values two weeks prior and two weeks after the missing value. However, if
+                            there are more missing records after interpolation than this limit (i.e. if a group of
+                            missing records is larger than --max-records-to-interpolate PLUS --max-records-to-impute)
+                            then the file will be discarded.
+    :param output_dir: The directory into which the generated EPW files will be written. If not passed, the files will
+        be written to a temporary location depending on the operating system the script is run in.
+    :param progress_handler: An optional function that, if passed, will be invoked for each file and will be passed
+        the file path about to be processed
+    :return: A tuple consisting of:
+        - A list of paths to all successfully generated EPW files
+        - A Dataframe with the columns "file" and "error" describing all files that could not be generated due to errors
+    """
+    # Initialize the df to hold paths of AMY files that could not be converted to an EPW.
+    errors = _pd.DataFrame(columns=['file', 'error'])
+
+    # Initialize the list of paths of successfully generated EPW files
+    amy_epw_file_paths = []
+
+    # If no output directory was specified, create a temporary directory
+    if output_dir is None:
+        output_dir = _tempfile.mkdtemp()
+
+    # Iterate through the set of AMY files, attempting to convert each one to an EPW file based on the
+    # corresponding TMY file
+    for amy_file_path in amy_file_paths:
+        if progress_handler is not None:
+            progress_handler(amy_file_path)
+
+        try:
+            amy_file_name = _os.path.basename(amy_file_path)
+            amy_file_name_without_ext = _os.path.splitext(amy_file_name)[0]
+
+            # Get the station number and year from the AMY file name
+            wmo_station_id = amy_file_name_without_ext.split("-")[0]
+            year = int(amy_file_name_without_ext.split("-")[-1])
+
+            # Get path to the TMY EPW file corresponding to that station.
+            tmy3_epw_file_path = _glob(
+                _os.path.join('..', 'inputs', 'Energy_Plus_TMY3_EPW', f'*.{wmo_station_id}_TMY3.epw')
```

Comment:
`inputs` will not, and should not, always be with this package.  It should be assumed the input dir to Energy_Plus... will be provided by the user.

Reply:
This has already been resolved in the csv_based_script branch, where these file paths are provided by the user.